pandas/core/strings.py

```diff
@@ -2466,7 +2474,27 @@ def cat(self, others=None, sep=None, na_rep=None, join="left"):
         2    -c
         dtype: object
 
-        For more examples, see :ref:`here <text.concatenate>`.
+        When `others` is a Series/Index/DataFrame, it is advisable
```

Comment:
you need to explain *why* one should do this. e.g. to have alignment since the join is defaulting to left.

show an example first with index and index, then index and series of the index and it should be more clear.